src/test/java/edu/cornell/kfs/gl/service/impl/ScheduledAccountingLineServiceImplTest.java

```diff
@@ -0,0 +1,121 @@
+package edu.cornell.kfs.gl.service.impl;
+
+import static org.junit.Assert.*;
+
+import java.sql.Date;
+import java.util.Calendar;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import edu.cornell.kfs.fp.CuFPConstants;
+import edu.cornell.kfs.fp.CuFPConstants.ScheduledSourceAccountingLineConstants.ScheduleTypes;
+import edu.cornell.kfs.fp.businessobject.ScheduledSourceAccountingLine;
+import edu.cornell.kfs.fp.service.impl.ScheduledAccountingLineServiceImpl;
+
+public class ScheduledAccountingLineServiceImplTest {
+	
+	ScheduledAccountingLineServiceImpl scheduledAccountingLineUtil;
+	Date startingPointDate;
+	Calendar startingPointCal;
+	ScheduledSourceAccountingLine accountingLine;
+
+	@Before
+	public void setUp() throws Exception {
+		scheduledAccountingLineUtil = new ScheduledAccountingLineServiceImpl();
+		
+		startingPointDate = new Date(Calendar.getInstance().getTimeInMillis());
+		startingPointCal = Calendar.getInstance();
+		startingPointCal.setTimeInMillis(startingPointDate.getTime());
+		
+		accountingLine = new ScheduledSourceAccountingLine();
+		accountingLine.setStartDate(startingPointDate);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		scheduledAccountingLineUtil = null;
+		startingPointDate = null;
+		startingPointCal = null;
+		accountingLine = null;
+	}
+	
+	private Date getDateResult(ScheduleTypes scheduleType, String count) {
+		accountingLine.setScheduleType(scheduleType.name);
+		accountingLine.setPartialTransactionCount(count);;
+		Date results = scheduledAccountingLineUtil.generateEndDate(accountingLine);
+		return results;
+	}
+	
+	private Date getExpectedDate(int dateAddElement, int addAmount) {
+		Calendar calcCal = startingPointCal.getInstance();
+		calcCal.add(dateAddElement, addAmount);
+		Date expectedDate = new Date(calcCal.getTimeInMillis());
+		return expectedDate;
+	}
+
+	@Test
+	public void testGenerateEndDateDaily() {
+		Date expectedDate = getExpectedDate(Calendar.DATE, 0);
```

Comment:
This is an improvement, but there are essentially the same three lines in each test method with only slight variations. It looks like you could put them in a method with four parameters and then each test method would only have one method call in it passing the appropriate parameters to the new method.